### ISSUE TITLE
Invalid credential

### ISSUE THREAD

**Issue by etkstim** [jabbink/PokemonGoBot#583]
Description:

I have insert in config.properties this line:

# PTC/Google login credentials
username=myemail@gmail.com

# Choose plaintext or base64'd password
password=mypassword
base64_password=

# Token to login directly
token=
latitude=39.8089707
longitude=16.1503722

I have error of invalid credentials, but my credentials are right

Version:
pogo.scraper-all-0.5.0-alpha2

**Operating System:**
Windows 8.1

**Java version:**
[Full output of `java -version`]
1.8.0_73b02

**Comment by rirze** [comment]
Have you tried logging into the app, and seeing if you are still able to access your account?

**Comment by etkstim** [comment]
Yes! I have able to log from app without problem :)

**Comment by pierobj** [comment]
Same issue, still doesn't work to me with Google Login (0.5.0-alpha2), it returns:

Server refused your login credentials. Are they correct?

Tried two different Google accounts and one PTC, the PTC works fine.

**Comment by etkstim** [comment]
I think that problem is two step authentication. Try to disable it and log on server

**Comment by pr0ves** [comment]
Is your token field empty? If not delete the token and retry

**Comment by LOLwhat123** [comment]
mine says get user to go to :null press :null what do i do?

**Comment by omar972** [comment]
@LOLwhat123 Servers are down atm, that might be the cause for that

**Comment by etkstim** [comment]
I have resolved problem with disable of two step auth

**Comment by SSminowjohnson** [comment]
go to :null press :null

This is a bug and will hopefully be fixed and reuploaded soon. That's all I get too and there is no token in my config and 2 step auth is turned off.

**Comment by pierobj** [comment]
I don't have 2 step auth and issue persists

**Comment by SSminowjohnson** [comment]
That's exactly what I wrote. It's not just you, it's a current issue in 0.4.3 and 0.4.2. Hopefully a fix is soon uploaded for those of us that don't wish to use 0.5.0.

**Comment by pierobj** [comment]
It's happening to me with 0.4.3, 0.5.0-alpha and 0.5.0-alpha2

**Comment by LOLwhat123** [comment]
my servers are still working and thanks man

**Comment by LOLwhat123** [comment]
is anyone actually botting on there main?

**Comment by omar972** [comment]
I am, but please try not to go off-topic.

**Comment by omar972** [comment]
If you want to use 2-factor authorization, you need to create an app password, not sure if this will work, but it should.
Go to [this](https://security.google.com/settings/u/0/security/apppasswords) and enter you email and password if need be. Expand drop down menu for 'Select app', select 'other', enter "Bot on windows" or whatever then press generate, make sure you copy it to clipboard.

Replace the password you are using in the config (Your main google password) with the password you just generated.

**Comment by LOLwhat123** [comment]
one last question where do i go to do :null lol

**Comment by omar972** [comment]
It's a bug, version 0.5.0 alpha 2 has the fix.

**Comment by LOLwhat123** [comment]
alpha 2 has a bug too lol

**Comment by omar972** [comment]
There is a reason it's alpha. Lol

**Comment by LOLwhat123** [comment]
true lol oh well i can wait :D whoever is coding it is doing a great job. PTC works perfect tho wish i could link my gmail to ptc

**Comment by tayjaf** [comment]
Getting #592 here.

**Comment by omar972** [comment]
@tayjaf 
Are you using 2 step authorization?

**Comment by tayjaf** [comment]
@omar972 No.

**Comment by omar972** [comment]
@tayjaf If you have a token in the config as well as your username and password, delete the username and password. If you don't have a token, you can generate one using curl or just use an older build of the bot to generate a token.

**Comment by SiriusCrypto** [comment]
same issue, unfortunately i have no idea how to generate a token using curl, i tried to google it but no luck

**Comment by omar972** [comment]
Well, google how to install curl and then use the command 
`curl "https://android.clients.google.com/auth" --data "Email=REPLACE_WITH_EMAIL&Passwd=REPLACE_WITH_PASSWORD&service=ac2dm&add_account=1"`
There should be a line "token=..................................." that would be it.

**Comment by ZacharyKipping** [comment]
For those of you not wanting to use curl (personally had issues with google's android auth link) and have cloned the source code you can replace the credentials property located on line 33 of the settings.kt with:

`credentials = GoogleAutoCredentials("YOUREMAILHERE", "YOURPASSWORDHERE"),`

This unfortunately was the only thing to resolve the problem of "not having the right credentials" with google authentication that I have been getting on all versions of this project. Currently only have errors with walkingloop occasionally, but that is a different matter entirely. Hope this helps some people.

**Comment by Killertom63** [comment]
I'm getting the same issue on Pokemon Account, not Google Account.

**Comment by zjkipping** [comment]
@Killertom63 If you are still having troubles and are willing to edit the source code, try what I commented earlier except for the PTC credentials:
`credentials = PtcCredentials("YOURUSERNAME", "YOUR PASSWORD"`
This might work as a potential workaround if you are cloning the source code. 100% works for google login for me (I just stash this code and update to develop periodically), I do not have a PTC login to test with though. Otherwise make 100% sure you are putting in the correct info for the PTC login. Hope this helps.

**Comment by Evuse** [comment]
@zjkipping still same problem for google login. I've changed like you sayd:


            credentials = GoogleAutoCredentials("mywmail", "mypassword"),

#if (properties.getProperty("username", "").isEmpty()) {
#                GoogleCredentials(properties.getProperty("token", ""))
#            } else if(properties.getProperty("username", "").contains("@")) {
#                GoogleAutoCredentials(properties.getProperty("username"), getPasswordProperty())
#            } else {
#                PtcCredentials(properties.getProperty("username"), getPasswordProperty())
#            },


but still having back this message:

31 Jul 22:44:29: Logging in to game server...
31 Jul 22:44:30: Server refused your login credentials. Are they correct?

**Comment by zjkipping** [comment]
@Evuse Make sure you do not have 2 step authorization for your google account or if you do get the proper password to bypass it, Also make sure you rebuilt the source code. After that I have no clue unfortunately, this seems to have worked for my set of issues.

**Comment by Killertom63** [comment]
@zjkipping yeah I have no idea how to do that. I'll just wait for a fix. Or go outside, haha. Thanks for the help though! :)

**Comment by zjkipping** [comment]
@Killertom63 Glady, the main page of the repository helps explain a lot towards getting everything working source code wise. If you do not want to download a repo client manager (which I suggest gitKraken or sourcetree), you can always download the source code in a zip which works the same. Other than that you just need to install Gradle and the Java SDK, I am not used to this dev enviroment myself but I used this tutorial to help:  http://bryanlor.com/blog/gradle-tutorial-how-install-gradle-windows and for the JDK: http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html (just accept the licence agreement and choose your OS for the 101 kit) and then run the commands the main repo page states.

If you don't want to do this I would understand though, glad I could help in a way.

**Comment by Cypherke** [comment]
have you tried alpha5?

**Comment by TheBronx** [comment]
alpha5 doesn't work either:
`01 ago 08:34:38: Logging in to game server...
01 ago 08:34:39: Server refused your login credentials. Are they correct?`
Same happens with an empty token, with an old token, with a curl token...

It is completely broken. App works, of course.

**Comment by Arihiza** [comment]
I only have this issue with Alpha 3. Everything else works fine.

**Comment by TheBronx** [comment]
v0.4.3 works "fine". logs in and starts to fail in profile and boot loops.

alphas just fail to log in. both in windows and linux. using precompiled jar version.

**Comment by Arihiza** [comment]
Got it working by disabling 2-step authentication like proposed here.

**Comment by jabbink** [comment]
@TheBronx Oracle or OpenJDK?

**Comment by TheBronx** [comment]
@jabbink Oracle
@Arihiza I don't have 2fa enabled.

**Comment by nerclid** [comment]
Same issue for me. I verified I am _not_ using 2 step auth, and I have logged into pokemon go app, and also to email account with same credentials. I tried putting in quotes, and no difference. I am using v0.5.0 precompiled on genuine windows 10 pro 64 bit, and jre-8u101-windows-x64.

I have not tried zjkipping's suggestion, as I am not sure/confident in my ability to do that properly.
